ggshield/core/extra_headers.py

```diff
@@ -0,0 +1,42 @@
+from typing import Dict, Optional
+
+import click
+
+
+def add_extra_header(
+    ctx: Optional[click.Context], key: str, value: str, overwrite: bool = False
+) -> bool:
+    """
+    Stores an extra header in the command's context.
+    Returns True if the extra headers are modified.
+
+    This function will put a dict in ctx.obj and store the headers in ctx.obj['headers']
+    """
+    if ctx is None:
+        return False
+
+    if not isinstance(ctx.obj, dict):
+        ctx.obj = {"headers": {}}
+    if "headers" not in ctx.obj:
+        ctx.obj["headers"] = {}
+
+    if key not in ctx.obj["headers"] or overwrite:
+        ctx.obj["headers"][key] = value
+        return True
+    return False
```

Comment:
I think this can be kept simpler: we don't need overwrite for now and we don't need the return value (current code does not use it). It can always be added later if needed.